app/services/ipam/base.rb

```diff
@@ -41,6 +41,18 @@ def serializable_hash(options = nil)
 
     private
 
+    def block_ip_cache_key(ip)
+      "blocked_ip_cache_#{subnet.network}/#{subnet.mask}__ip_#{ip}"
```

Comment:
I think you can simply use subnet's id, no need to use network/mask. Then concatenate the IP address.

Out of curiosity what is the reason for those `_` or double `__` characters? It will eventually get hashed, you can just use slash for all strings.

Reply:
I just thought it makes it more readable. I fix it.